File: src/main/java/leetcode/easy/Problem9.java
package leetcode.easy;

public class Problem9 {
  public boolean isPalindrome(int x) {
    if (x == 0) {
      return true;
    }
    if (x < 0) {
      return false;
    }

    int n = x;
    int cnt = 0;
    while (n != 0) {
      n /= 10;
      cnt += 1;
    }

    n = x;
    int[] arr = new int[cnt];
    for (int i=0; i<cnt; i++) {
      arr[cnt-1-i] = n%10;
      n /= 10;
    }

    for (int i=0; i<=cnt/2; i++) {
      if (arr[i] != arr[cnt-1-i]) {
        return false;
      }
    }
    return true;
  }
}

File: src/main/java/leetcode/easy/Problem9.kt
package leetcode.easy

class Problem9 {
    fun isPalindrome(x: Int): Boolean {
        if (x == 0) return true
        if (x < 0) return false

        var cnt = 0
        var n: Int = x
        while (n != 0) {
            cnt += 1
            n /= 10
        }

        n = x;
        val arr = IntArray(cnt)
        for (i in 0..<cnt) {
            arr[cnt - 1 - i] = n % 10
            n /= 10
        }

        for (i in 0..<cnt) {
            if (arr[i] != arr[cnt - 1 - i]) return false
        }

        return true
    }
}